content/using_ec2_spot_instances_with_eks/spotworkers/deployhandler.md

```diff
@@ -0,0 +1,67 @@
+---
+title: "Deploy The Spot Interrupt Handler"
+date: 2018-08-07T12:32:40-07:00
+weight: 40
+draft: false
+---
+
+When users requests On-Demand instances from a pool to the point that the pool is depleted, the system will select a set of spot instances from the pool to be terminated. A Spot instance pool is a set of unused EC2 instances with the same instance type (for example, m5.large), operating system and Availability Zone. The Spot Instance is sent an interruption notice two minutes ahead to gracefully wrap up things. 
+
+We will deploy a pod on each spot instance to detect the instance termination notification signal so that we can both terminate gracefully any pod that was running on that node, drain from load balancers and redeploy applications elsewhere in the cluster.
+
+To deploy Spot Interrupt Handler on each Spot Instance we will use a [DaemonSet](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/). This will monitor the EC2 metadata service on the instance for a interruption notice.
+
+Within the Spot Interrupt Handler DaemonSet, the workflow can be summarized as:
+
+* Identify that a Spot Instance is being reclaimed.
+* Use the 2-minute notification window to gracefully prepare the node for termination.
+* [**Taint**](https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/) the node and cordon it off to prevent new pods from being placed.
+* [**Drain**](https://kubernetes.io/docs/tasks/administer-cluster/safely-drain-node/) connections on the running pods.
+* Replace the pods on remaining nodes to maintain the desired capacity.
+
+We have provided an example K8s DaemonSet manifest. A DaemonSet runs one pod per node.
+
+```
+mkdir ~/environment/spot
+pushd ~/environment/spot
+wget https://raw.githubusercontent.com/ruecarlo/ec2-spot-workshops/eks_workshop/content/using_ec2_spot_instances_with_eks/spotworkers/deployhandler.files/spot-interrupt-handler-example.yml
```

Comment:
before you used `curl --silent`

Reply:
Thinking it even more , not inclined on adding the `--silent`. This shows progress on the download. I think it's useful, for troubleshooting as well on places where connectivity might not be great.